components/calendar/calendar.js

```diff
@@ -0,0 +1,480 @@
+	return (month === 11) ? 0 : (month + 1);
+}
+
+function getNumberOfDaysFromPrevMonthToShow(month, year, firstDayOfWeek) {
+	const firstDayOfMonth = new Date(year, month, 1).getDay();
+	let numDaysFromLastMonthToShowThisMonth = 0;
+	if (firstDayOfMonth !== firstDayOfWeek) {
+		numDaysFromLastMonthToShowThisMonth = firstDayOfMonth - firstDayOfWeek;
+		if (numDaysFromLastMonthToShowThisMonth < 0) {
+			numDaysFromLastMonthToShowThisMonth += daysInWeek;
+		}
+	}
+	return numDaysFromLastMonthToShowThisMonth;
+}
+
+function getNumberOfDaysInMonth(month, year) {
+	month++; // month index started at 0
+	let days = 31;
+	if (month === 2) {
+		if ((year % 4 === 0) && ((year % 100 !== 0) || (year % 400 === 0))) {
+			days = 29;
+		} else {
+			days = 28;
+		}
+	} else if (month === 4 || month === 6 || month === 9 || month === 11) {
+		days = 30;
+	}
+	return days;
+}
+
+function getNumberOfDaysToSameWeekPrevMonth(month, year, firstDayOfWeek) {
+	const prevMonth = getPrevMonth(month);
+	const prevMonthYear = prevMonth === 11 ? year - 1 : year;
+	const numDaysPrevMonth = getNumberOfDaysInMonth(prevMonth, prevMonthYear);
+	const numDaysFromPrevPrevMonthToShowPrevMonth = getNumberOfDaysFromPrevMonthToShow(prevMonth, prevMonthYear, firstDayOfWeek);
+	const numWeeksPrevMonth = Math.ceil((numDaysFromPrevPrevMonthToShowPrevMonth + numDaysPrevMonth) / 7);
+
+	const firstDayOfMonth = new Date(year, month, 1).getDay();
+	const hasDaysFromPrevMonth = (firstDayOfMonth !== firstDayOfWeek);
+	return daysInWeek * (numWeeksPrevMonth - ((hasDaysFromPrevMonth) ? 1 : 0));
+}
+
+function getPrevMonth(month) {
+	return (month === 0) ? 11 : (month - 1);
+}
+
+class Calendar extends LocalizeStaticMixin(LitElement) {
+
+	static get properties() {
+		return {
+			selectedValue: { type: String, attribute: 'selected-value' },
+			summary: { type: String },
+			_shownMonth: { type: Number }
+		};
+	}
+
+	static get styles() {
+		return [calendarStyles, bodySmallStyles, heading4Styles];
+	}
+
+	static get resources() {
+		return {
+			'ar': { show: 'Show {month} ' },
+			'en': { show: 'Show {month}' },
+			'es': { show: 'Show {month} ' },
+			'fr': { show: 'Show {month} ' },
+			'ja': { show: 'Show {month} ' },
+			'ko': { show: 'Show {month} ' },
+			'nl': { show: 'Show {month} ' },
+			'pt': { show: 'Show {month} ' },
+			'sv': { show: 'Show {month} ' },
+			'tr': { show: 'Show {month} ' },
+			'zh': { show: 'Show {month} ' },
+			'zh-tw': { show: 'Show {month} ' }
+		};
+	}
+
+	constructor() {
+		super();
+
+		this._descriptor = getDateTimeDescriptor();
+
+		this._daysOfWeekIndex = [];
+		for (let i = this._descriptor.calendar.firstDayOfWeek; i < this._descriptor.calendar.firstDayOfWeek + daysInWeek; i++) {
+			this._daysOfWeekIndex.push(i % daysInWeek);
+		}
+
+		this._dialogLabelId = getUniqueId();
+	}
+
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+
+		this.selectedValue = this.selectedValue ? new Date(this.selectedValue) : null;
```

Comment:
Agreed. We could have a separate `_selectedValueDate` that's the `Date`-ified version of `selectedValue` (since it's useful for internal logic) and keep `this.selectedValue` as the string value?

Reply:
Yeah you could, although then you need to always keep those 2 things in sync whenever something internal or external updates the property. In the time-input, we just created a parse function that converts the string to a `Date` and we just call it whenever we need the object version.